### PR TITLE
Add resource requirements for vcheck container

### LINKED ISSUE #796: cockroachdb-vcheck pod inherits resource limits and requests from cluster nodes
We faced an issue during cluster upgrade when `cockroachdb-vcheck` pod stuck in `Pending` state:

```
NAME                                                              READY   STATUS    RESTARTS   AGE
cockroach-operator-58996d89f9-xqlzn                               1/1     Running   1          14h
cockroachdb-0                                                     1/1     Running   0          8d
cockroachdb-1                                                     1/1     Running   0          8d
cockroachdb-2                                                     1/1     Running   0          8d
cockroachdb-vcheck-27248336-4tdpj                                 0/1     Pending   0          100
```

The describe shows that it got Resource Limits and Requests set to the values which was defined for cluster nodes in Custom Resource:

```
$ kubectl describe po cockroachdb-vcheck-27248332-ncb4k

...
    Limits:
      cpu:     6
      memory:  24Gi
    Requests:
      cpu:        4
      memory:     24Gi
    Environment:  <none>
    Mounts:       <none>
Conditions:
  Type           Status
  PodScheduled   False
Volumes:         <none>
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     dedicated=redis:NoSchedule
                 dedicated=redis:NoExecute
                 node.kubernetes.io/not-ready:NoExecute op=Exists for 600s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 600s

Events:
  Type     Reason            Age                 From               Message
  ----     ------            ----                ----               -------
  Warning  FailedScheduling  55s (x3 over 2m3s)  default-scheduler  0/23 nodes are available: 2 node(s) had taint {dedicated: alb}, that the pod didn't tolerate, 2 node(s) had taint {dedicated: links-varnish}, that the pod didn't tolerate, 2 node(s) had taint {dedicated: links}, that the pod didn't tolerate, 2 node(s) had taint {dedicated: location}, that the pod didn't tolerate, 2 node(s) had taint {dedicated: varnish}, that the pod didn't tolerate, 4 node(s) had taint {dedicated: edge}, that the pod didn't tolerate, 5 Insufficient memory, 9 Insufficient cpu.
```

Due to this fact and lack of one more big worker node in our k8s cluster we can't perform cluster upgrade.

Here is our `CrdbCluster` resource configuration:

```
image:
  pullSecret: icm-coreeng-pull-secret
nodeSelector:
  dedicated: redis
nodes: 3
persistence:
  accessModes:
  - ReadWriteOnce
  enabled: true
  storage: 20Gi
  storageClassName: ibmc-block-bronze
  volumeMode: Filesystem
resources:
  limits:
    cpu: "6"
    memory: 24Gi
  requests:
    cpu: "4"
    memory: 24Gi
tlsEnabled: true
tolerations:
- effect: NoSchedule
  key: dedicated
  operator: Equal
  value: redis
- effect: NoExecute
  key: dedicated
  operator: Equal
  value: redis
```

Does the `cockroachdb-vcheck` require the same Resource Requests as cluster nodes or it should run with less resources and this is a bug?

```
k8s v1.22.2
cockroachdb-operator:v2.3.0
cockroach:v21.1.10
```

### PR DESCRIPTION
**Pull request by antonioua**
Added resource requirements for vcheck container.
Resolves #796

### CODE REVIEW

**Comment by antonioua** [comment]
@pseudomuto , I've hardcoded resource requirements, please take a look, let's keep it simple.

File: pkg/resource/job.go [hunk 1]
```diff
@@ -126,14 +125,24 @@ func (b JobBuilder) MakeContainers() []corev1.Container {
...
 	return []corev1.Container{
 		{
 			Name:            JobContainerName,
 			Image:           b.GetCockroachDBImageName(),
 			ImagePullPolicy: *b.Spec().Image.PullPolicyName,
-			Resources:       b.Spec().Resources,
-			Command:         []string{"/bin/bash"},
-			Args:            []string{"-c", fmt.Sprintf("%s; sleep 150", GetTagVersionCommand)},
+			Resources: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
```
Comment (pseudomuto) [inline pkg/resource/job.go:135]: We can likely make these the same as the requests. This simply runs `cockroach version` and won't require the ability to scale vertically IMO

**Review by pseudomuto — APPROVED** [review]
Checked it and ran a test locally. Works like a charm 👍

**Review by davidwding — APPROVED** [review]

**Comment by antonioua** [comment]
I've increased resource requests so vcheck passes from 1st try.

**Comment by pseudomuto** [comment]
Hey @antonioua, can you rebase this on the latest master? Then we'll run the tests and get this merged

**Comment by antonioua** [comment]
@pseudomuto , I've done rebase.

**Pull request merged**